File: src/main/java/net/xalcon/analyzeio/AnalyzeIO.java
package net.xalcon.analyzeio;

import net.minecraftforge.fml.common.Mod;
import net.minecraftforge.fml.common.Mod.EventHandler;
import net.minecraftforge.fml.common.event.FMLInitializationEvent;
import net.minecraftforge.fml.common.event.FMLPreInitializationEvent;
import net.minecraftforge.fml.common.network.NetworkRegistry;
import net.xalcon.analyzeio.client.GuiHandler;
import org.apache.logging.log4j.Logger;

@Mod(modid = AnalyzeIO.MODID, name = AnalyzeIO.NAME, version = AnalyzeIO.VERSION)
public class AnalyzeIO
{
    public static final String MODID = "analyzeio";
    public static final String NAME = "AnalyzeIO";
    public static final String VERSION = "0.4";

    private static Logger logger;
    private static AnalyzeIO instance;

    private AnalyzeIO() { }

    @Mod.InstanceFactory
    public static AnalyzeIO getInstance()
    {
        if(instance == null)
            instance = new AnalyzeIO();
        return instance;
    }

    @EventHandler
    public void preInit(FMLPreInitializationEvent event)
    {
        logger = event.getModLog();
    }

    @EventHandler
    public static void init(FMLInitializationEvent event)
    {
        NetworkRegistry.INSTANCE.registerGuiHandler(getInstance(), new GuiHandler());
    }
}

File: src/main/java/net/xalcon/analyzeio/AnalyzeIO.kt
package net.xalcon.analyzeio

import net.minecraftforge.fml.common.Mod
import net.minecraftforge.fml.common.SidedProxy
import net.minecraftforge.fml.common.event.FMLInitializationEvent
import net.minecraftforge.fml.common.event.FMLPreInitializationEvent
import net.minecraftforge.fml.common.network.NetworkRegistry
import net.xalcon.analyzeio.common.CommonProxy
import net.xalcon.analyzeio.common.GuiHandler
import net.xalcon.analyzeio.common.init.ModBlocks
import net.xalcon.analyzeio.common.init.ModItems
import net.xalcon.analyzeio.common.init.ModRecipes
import kotlin.jvm.JvmStatic

@Mod(
        modid = AnalyzeIO.MODID,
        name = AnalyzeIO.MODNAME,
        version = AnalyzeIO.VERSION,
        dependencies = AnalyzeIO.DEPENDENCIES,
        acceptedMinecraftVersions = AnalyzeIO.ACCEPTED_MC_VERSIONS
)
object AnalyzeIO
{
    const val MODID = "analyzeio"
    const val MODNAME = "AnalyzeIO"
    const val VERSION = "@VERSION@"
    const val DEPENDENCIES = "required-after:forgelin@[1.4.2,);required-after:EnderIO@[1.10.2-3.1.183,);"
    const val ACCEPTED_MC_VERSIONS = "[1.10.2]"
    const val CLIENT_PROXY = "net.xalcon.analyzeio.client.ClientProxy"
    const val SERVER_PROXY = "net.xalcon.analyzeio.common.CommonProxy"

    @SidedProxy(clientSide = CLIENT_PROXY, serverSide = SERVER_PROXY)
    lateinit var PROXY: CommonProxy

    @Mod.EventHandler
    fun preInit(event: FMLPreInitializationEvent)
    {
        ModBlocks.init()
        ModItems.init()
    }

    @Mod.EventHandler
    fun init(event: FMLInitializationEvent)
    {
        PROXY.init(event)
        ModRecipes.init()
        NetworkRegistry.INSTANCE.registerGuiHandler(this, GuiHandler)
    }

    @JvmStatic
    @Mod.InstanceFactory
    fun createModInstance() = this
}